### ISSUE TITLE
other backends?

### ISSUE THREAD

**Issue by allefeld** [has2k1/plotnine#502]
Does plotnine have an abstraction level that would make it (relatively) easy to implement backends besides matplotlib? I'm asking because I really like the gg logic, but would prefer interactive graphics like those provided by Plotly or Bokeh.

**Comment by has2k1** [comment]
Plotnine does not have an abstraction level nonetheless, another backend is possible. How easy? It would depend on the package.                                  
                        .

**Comment by TyberiusPrime** [comment]
Theres also  this 'matplotlib->d3' conversion library,
which I presume should work with plotnine - I haven't tried it though.

https://mpld3.github.io/index.html

**Comment by jdanbrown** [comment]
@has2k1 Have you ever thought about adding a vega-lite backend like altair?
- I find that altair renders plots _multiple_ times faster than plotnine (e.g. 25s -> 7s in a big facet_grid plot I happen to be working on right now) — I'd guess this is because plotnine is generating a big png (~1900x1000px) whereas altair/vega are just generating js/dom for the browser to render
- But I much prefer the ergonomics of plotnine/ggplot code over the corresponding altair code (e.g. time histograms have weird challenges in altair/vega), as well as the aesthetics of plotnine/ggplot plots over altair plots (e.g. facets strips in plotnine/ggplot are visually prominent whereas in altair they aren't prominent and get noised out by the repeated axis titles)
- (Aside: the main thing I _would_ miss from altair is the flexibility of combining arbitrary plots with vconcat/hconcat. I'd love to have that capability in plotnine, but that's a separate issue.)

I'd love to have the best of both worlds! plotnine + vega-lite = <3

**Comment by has2k1** [comment]
Yes I have thought about vega-lite, but I do not have the bandwidth at the to take on javascript.

**Comment by ahll19** [comment]
Are there any updates on this issue? I am trying to find workarounds on the internet but there aren't many to be found, and what can be found doesn't work or is too janky.
